source/cgal/intersections.cc

```diff
@@ -0,0 +1,737 @@
+                                 triangle[2])}};
+
+                            vertices.push_back(verts);
+                          }
+                      }
+                  }
+              }
+          }
+      }
+
+    return vertices;
+
+#  else
+
+    Assert(
+      false,
+      ExcMessage(
+        "This function requires a version of CGAL greater or equal than 5.1.5."));
+    (void)cell0;
+    (void)cell1;
+    (void)mapping0;
+    (void)mapping1;
+    (void)tol;
+    return {};
+#  endif
+  }
+
+
+  template <>
+  std::vector<std::array<Point<3>, 4>>
+  compute_intersection_of_cells<3, 3, 3>(
+    const typename Triangulation<3, 3>::cell_iterator &cell0,
+    const typename Triangulation<3, 3>::cell_iterator &cell1,
+    const Mapping<3, 3> &                              mapping0,
+    const Mapping<3, 3> &                              mapping1,
+    const double                                       tol)
+  {
+    Surface_mesh surf0, surf1, sm;
+    CGALWrappers::dealii_cell_to_cgal_surface_mesh(cell0, mapping0, surf0);
+    CGALWrappers::dealii_cell_to_cgal_surface_mesh(cell1, mapping1, surf1);
+    CGAL::Polygon_mesh_processing::triangulate_faces(surf0);
+    CGAL::Polygon_mesh_processing::triangulate_faces(surf1);
+    CGALWrappers::compute_boolean_operation(
+      surf0, surf1, CGALWrappers::BooleanOperation::compute_intersection, sm);
+    if (CGAL::Polygon_mesh_processing::volume(sm) > tol)
+      {
+        // Collect tetrahedrons
+        std::vector<std::array<Point<3>, 4>> vertices;
+        Triangulation3_inexact               tria;
+        tria.insert(sm.points().begin(), sm.points().end());
+        for (const auto &c : tria.finite_cell_handles())
+          {
+            const auto &tet = tria.tetrahedron(c);
+            vertices.push_back(
+              {{CGALWrappers::cgal_point_to_dealii_point<3>(tet.vertex(0)),
+                CGALWrappers::cgal_point_to_dealii_point<3>(tet.vertex(1)),
+                CGALWrappers::cgal_point_to_dealii_point<3>(tet.vertex(2)),
+                CGALWrappers::cgal_point_to_dealii_point<3>(tet.vertex(3))}});
+          }
+        return vertices;
+      }
+    else
+      {
+        return {};
+      }
+  }
+
+} // namespace CGALWrappers
+
+DEAL_II_NAMESPACE_CLOSE
+
+#else
+
+DEAL_II_NAMESPACE_OPEN
+
+template <int dim0, int dim1, int spacedim>
+std::vector<std::array<Point<spacedim>, dim1 + 1>>
+compute_intersection_of_cells(
+  const typename Triangulation<dim0, spacedim>::cell_iterator &cell0,
+  const typename Triangulation<dim1, spacedim>::cell_iterator &cell1,
+  const Mapping<dim0, spacedim> &                              mapping0,
+  const Mapping<dim1, spacedim> &                              mapping1,
+  const double                                                 tol = 1e-9)
+{
+  (void)cell0;
+  (void)cell1;
+  (void)mapping0;
+  (void)mapping1;
+  (void)tol;
+  AssertThrow(false, ExcMessage("This function needs CGAL to be installed."));
```

Comment:
Could you add an exception like https://github.com/dealii/dealii/blob/75eeea84ed5c7a888cc257e822c56a7e59cabb98/include/deal.II/base/exceptions.h#L1157-L1158?